expression/aggregation/base_func.go

```diff
@@ -184,6 +184,10 @@ func (a *baseFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		a.Args[0] = expression.BuildCastFunction(ctx, a.Args[0], tp)
 	}
 	a.RetTp = a.Args[0].GetType()
+	if (a.Name == ast.AggFuncMax || a.Name == ast.AggFuncMin) && !a.RetTp.Hybrid() {
+		a.RetTp = a.Args[0].GetType().Clone()
+		a.RetTp.Flag ^= mysql.NotNullFlag
```

Comment:
`^=` is xor. It will filp that bit instead of resetting that bit.

Reply:
@AndrewDi If we reset NotNullFlag instead of using xor, HybridType is not needed to be checked?

Reply:
OK, change to `a.RetTp.Flag &^= mysql.NotNullFlag`

Reply:
@XuHuaiyu `a.RetTp = a.Args[0].GetType().Clone()` will break current Hybird type process.

Reply:
I do not really understand what's your meaning. @AndrewDi

Reply:
@XuHuaiyu TypeBit should always not null, so we should not reset flag for this type.